src/hooks/useOnOutsideClick.tsx

```diff
@@ -0,0 +1,59 @@
+import React, {useEffect, useCallback} from 'react'
+
+export type TouchOrMouseEvent = MouseEvent | TouchEvent
+
+export type UseOnOutsideClickProps = {
+  containerRef: React.RefObject<HTMLDivElement>
+  ignoreClickRefs?: React.RefObject<HTMLElement> []
+  onClickOutside: (e: TouchOrMouseEvent) => void
+}
+
+type ShouldCallClickHandlerProps = {
+  ignoreClickRefs?: React.RefObject<HTMLElement> []
+  containerRef: React.RefObject<HTMLDivElement>
+  e: TouchOrMouseEvent
+}
+
+const shouldCallClickHandler = ({ignoreClickRefs, containerRef, e}: ShouldCallClickHandlerProps) : boolean => {
```

Comment:
Going with the above comment about `Props`, I would either rename this to `ShouldCallClickHandlerArgs`, or, since there are only 3 here, it might be better to just make them 3 separate function arguments. Since this is not exported it's of low importance to me.